packages/react/script/prop-docs/build.ts

```diff
@@ -0,0 +1,149 @@
+const docgen = require('react-docgen-typescript')
+const globby = require('globby')
+
+const fs = require('node:fs')
+
+const docgenOptions = {
+  // savePropValueAsString: true,
+  // allowSyntheticDefaultImports might let us avoid separately importing types from React
+  // (e.g.: you can do `React.FC` instead of importing and using `FC` from React)
+  // allowSyntheticDefaultImports: true,
+  propFilter: (prop, _component) => {
+    if (prop.declarations !== undefined && prop.declarations.length > 0) {
+      const hasPropAdditionalDescription = prop.declarations.find(declaration => {
+        return !declaration.fileName.includes('node_modules')
+      })
+
+      return Boolean(hasPropAdditionalDescription)
+    }
+
+    return true
+  },
+}
+
+// const noPropsWhitelist = [
+//   'ButtonGroup/ButtonGroup.tsx',
+//   {fileName: 'ConfirmationDialog/ConfirmationDialog.tsx', displayName: 'useConfirm'},
+//   'UnderlineNav/LoadingCounter.tsx',
+//   {fileName: 'components/LiveRegion.tsx', displayName: 'LiveRegion'},
+//   {fileName: 'DataTable/Table.tsx', displayName: 'TableActions'},
+//   {fileName: 'DataTable/Table.tsx', displayName: 'TableHead'},
+//   {fileName: 'DataTable/Table.tsx', displayName: 'TableBody'},
+//   {fileName: 'DataTable/Table.tsx', displayName: 'TableRow'},
+// ]
+
+// Parse a file for docgen info
+const files = globby.sync(
+  // Glob for testing
+  // [
+  //   './packages/react/src/Avatar/*.tsx',
+  //   './packages/react/src/DataTable/*.tsx',
+  //   '!./packages/react/src/**/*.stories.tsx',
+  //   '!./packages/react/src/**/*.test.tsx',
+  // ],
+  ['./packages/react/src/**/*.tsx', '!./packages/react/src/**/*.stories.tsx', '!./packages/react/src/**/*.test.tsx'],
+  {
+    absolute: true,
+  },
+)
+const docgenOutput = docgen.parse(files, docgenOptions)
+
+const printSkippedComponents = () => {
+  console.log('COMPONENTS SKIPPED:')
+
+  files.forEach(file => {
+    if (!docgenOutput.map(({filePath}) => filePath).includes(file)) {
+      console.log(file)
+    }
+  })
+
+  console.log('\n')
+}
+
+const printSkippedProps = () => {
+  console.log('PROPS NOT DOCUMENTED:')
+
+  docgenOutput.forEach(({props, filePath, displayName}) => {
+    if (Object.keys(props).length === 0) {
+      console.log(`${filePath} - ${displayName}`)
+    }
+  })
+
+  console.log('\n')
+}
+
+const formatComponentJson = ({description, displayName, props, tags}) => {
+  const {alias, primerid, primerstatus, primera11yreviewed, primerstories = '', primerparentid} = tags
+
+  return {
+    id: primerid, // TODO: consider auto-generating an ID if one is not present by parsing `displayName`
```

Comment:
Just FYI @broccolinisoup and I just worked up [this PR](https://github.com/primer/design/pull/751) and [this PR](https://github.com/primer/react/pull/4458) that introduce a `docsId` field for grouping related components that have different names, eg. `tooltip` and `tooltip_v2`. We'll probably want to generate that field in this script too, although I'm not sure where the data would come from.